docs/docs/recipes/working-with-plugins.md

```diff
@@ -0,0 +1,101 @@
+
+## Using a plugin
+
+Found a plugin you'd like to use on your project? Awesome! You can configure it for use by following the steps below. This recipe uses the [`gatsby-source-filesystem` plugin](/plugins/gatsby-source-filesystem) as the example because it is one of the most popular commonly used plugins.
+
+> If you'd like to take a look at the available plugins you can install, check out the [plugin library](/plugins).
+
+### Prerequisites
+
+- An existing [Gatsby site](/docs/quick-start/) with a `gatsby-config.js` file
+
+### Directions
+
+1. Install the `gatsby-source-filesystem` plugin by running the following command:
+
+```shell
+npm install gatsby-source-filesystem
+```
+
+2. Add the plugin to your `gatsby.config.js`, and set any options it needs, the filesystem source plugin takes a `name` and `path` as options:
+
+```javascript:title=gatsby-config.js
+module.exports = {
+  plugins: [
+    // highlight-start
+    {
+      resolve: `gatsby-source-filesystem`,
+      options: {
+        name: `images`,
+        path: `${__dirname}/src/images`,
+      },
+    },
+    // highlight-end
+  ],
+}
+```
+
+_The instructions found in the README of the plugin you're using can you help you determine specifics about what configurations (if any) it requires. For this example to have an effect in your own site, you would need a folder of files at `src/images`._
+
+3. Run `gatsby develop`, your plugin should run as your site builds.
+
+### Additional resources
+
+- Learn more about configuring options or using default options in the [Using a Plugin in Your Site](/docs/using-a-plugin-in-your-site/) guide
+- See an example Gatsby site using this configuration in [the repo for the default Gatsby starter](https://github.com/gatsbyjs/gatsby-starter-default/blob/master/gatsby-config.js)
+
+## Creating a new plugin using a plugin starter
+
+If you want to create your own plugin with the minimal boilerplate Gatsby looks for in a plugin, you can get one started with the Gatsby plugin starter.
+
+### Prerequisites
+
+- An existing [Gatsby site](/docs/quick-start/) with a `gatsby-config.js` file
+
+### Directions
+
+1. _Outside_ of your Gatsby site, generate a new plugin based on the starter using the `gatsby new` command:
+
+```shell
+gatsby new my-plugin https://github.com/gatsbyjs/gatsby-starter-plugin
+```
+
+The directory structure should look something like this:
+
+```
+gatsby-site
+└── gatsby-config.js
+└── src
+    ├── components
+    └── pages
+my-plugin
+├── gatsby-browser.js
+├── gatsby-node.js
+├── gatsby-ssr.js
+├── index.js
+└── package.json
+```
+
+2. Add the plugin to your site's `gatsby-config.js`, linking it your local plugin's root folder with `require.resolve`:
```

Comment:
```suggestion
2. Add the plugin to your site's `gatsby-config.js`, linking it to your local plugin's root folder with `require.resolve`:
```
Worth mentioning that the name matches the directory name you used when generating it?

Reply:
Yes, I'll add that!!